src/Lecture1.hs

```diff
@@ -98,7 +102,8 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> [Char] -> [Char]
+subString start end str | end < 0 = "" | start < 0 && end >= 0 = take (end + 1) str | otherwise = take (end - start + 1) (drop start str)
```

Comment:
To make code easier to read, it's usually recommended to put guards on separate lines:
```suggestion
subString start end str
    | end < 0 = ""
    | start < 0 && end >= 0 = take (end + 1) str
    | otherwise = take (end - start + 1) (drop start str)
```